host/kinesis.go

```diff
@@ -0,0 +1,82 @@
+package host
+
+import (
+	"context"
+	"log/slog"
+	"time"
+
+	gk "github.com/dennis-tra/go-kinesis"
+	"github.com/probe-lab/hermes/tele"
+)
+
+type KinesisDataStream struct {
+	producer *gk.Producer
+	ctx      context.Context
+	cancelFn context.CancelFunc
+}
+
+var _ DataStream = (*KinesisDataStream)(nil)
+
+// NewKinesisDataStream creates a new instance of KinesisDataStream with a given producer.
+func NewKinesisDataStream(p *gk.Producer) *KinesisDataStream {
+	return &KinesisDataStream{
+		producer: p,
+		ctx:      nil,
+		cancelFn: nil,
+	}
+}
+
+// Type returns the type of the data stream
+func (k *KinesisDataStream) Type() DataStreamType {
+	return DataStreamTypeKinesis
+}
+
+// Start begins the data stream's operation.
+func (k *KinesisDataStream) Start(ctx context.Context) error {
+	dsCtx, dsCancel := context.WithCancel(ctx)
+
+	k.ctx = dsCtx
+	k.cancelFn = dsCancel
+
+	if err := k.producer.Start(ctx); err != nil {
+		return err
+	}
+
+	<-dsCtx.Done()
+
+	return dsCtx.Err()
+}
+
+// Stop ends the data stream.
+func (k *KinesisDataStream) Stop(ctx context.Context) error {
+	// wait until the producer has stopped
+	timeoutCtx, timeoutCncl := context.WithTimeout(k.ctx, 15*time.Second)
+	if err := k.producer.WaitIdle(timeoutCtx); err != nil {
+		slog.Info("Error waiting for producer to become idle", tele.LogAttrError(err))
+	}
+
```

Comment:
`staticcheck` doesn't seem to be happy with this `timeoutCncl`, this should make the tests happy
```suggestion
	timeoutCncl()
```